var/spack/repos/builtin/packages/tpl-builder/package.py

```diff
@@ -0,0 +1,144 @@
+
+    depends_on("hypre~shared", when="~shared+hypre")
+    depends_on("hypre+shared", when="+shared+hypre")
+    depends_on("blas", when="+lapack")
+    depends_on("lapack", when="+lapack")
+
+    requires("+lapack", when="+hypre")
+
+    for _flag in list(CudaPackage.cuda_arch_values):
+        depends_on("hypre cuda_arch=" + _flag, when="+hypre+cuda cuda_arch=" + _flag)
+        depends_on("kokkos cuda_arch=" + _flag, when="+kokkos+cuda cuda_arch=" + _flag)
+
+    for _flag in ROCmPackage.amdgpu_targets:
+        depends_on("hypre amdgpu_target=" + _flag, when="+hypre+rocm amdgpu_target=" + _flag)
+        depends_on("kokkos amdgpu_target=" + _flag, when="+kokkos+rocm amdgpu_target=" + _flag)
+
+    # MPI related dependencies
+    depends_on("mpi", when="+mpi")
+
+    phases = ["cmake", "build"]
+
+    def cmake_args(self):
+        spec = self.spec
+
+        options = [
+            self.define("INSTALL_DIR", spec.prefix),
+            self.define("DISABLE_ALL_TESTS", True),
+            self.define("CXX_STD", "17"),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant("ENABLE_SHARED", "shared"),
+            self.define("ENABLE_STATIC", not spec.variants["shared"].value),
+            self.define_from_variant("USE_MPI", "mpi"),
+            self.define("MPI_SKIP_SEARCH", False),
+            self.define_from_variant("USE_OPENMP", "openmp"),
+            self.define("DISABLE_GOLD", True),
+        ]
+
+        if spec.satisfies("+cuda"):
+            cuda_arch = spec.variants["cuda_arch"].value
+            cuda_flags = ["-extended-lambda", "--expt-relaxed-constexpr"]
+            if cuda_arch[0] != "none":
+                options.extend(
+                    [
+                        self.define("USE_CUDA", True),
+                        self.define(
+                            "CMAKE_CUDA_COMPILER", join_path(spec["cuda"].prefix.bin, "nvcc")
+                        ),
+                        self.define("CMAKE_CUDA_ARCHITECTURES", cuda_arch),
+                        self.define("CMAKE_CUDA_FLAGS", " ".join(cuda_flags)),
+                    ]
+                )
+
+        if spec.satisfies("+rocm"):
+            amdgpu_target = spec.variants["amdgpu_target"].value
+            if amdgpu_target[0] != "none":
+                options.extend(
+                    [
+                        self.define("USE_HIP", True),
+                        self.define(
+                            "CMAKE_HIP_COMPILER",
+                            join_path(spec["llvm-amdgpu"].prefix.bin, "amdclang++"),
+                        ),
+                        self.define("CMAKE_HIP_ARCHITECTURES", amdgpu_target),
+                        self.define("CMAKE_HIP_FLAGS", ""),
+                    ]
+                )
+
+        tpl_list = []
+
+        if spec.satisfies("+lapack"):
+            tpl_list.append("LAPACK")
+            if spec.satisfies("^intel-mkl"):
```

Comment:
Specify which virtual dependency this applies to.
```suggestion
            if spec.satisfies("^[virtuals=lapack] intel-mkl"):
```
Similar below.

Reply:
Fixed.